sdk/core/Azure.Core.Experimental/tests/public/JsonDataArrayTests.cs

```diff
@@ -0,0 +1,294 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests.Public
+{
+    public class JsonDataArrayTests
+    {
+        #region Convert tests
+
+        [Test]
+        public void CanConvertArrayToIEnumerable()
+        {
+            dynamic data = JsonDataTestHelpers.CreateFromJson("[1, 2, 3]");
+
+            // TODO: This is interesting - what is commented out below fails,
+            // because apparently the type system has decided the IEnumerable
+            // is holding JsonData, not dynamic, so the DMO callback is never called
+            // and the explicit cast to int is not visible b/c JsonData is internal.
+            //IEnumerable value = (IEnumerable)data;
```

Comment:
I believe this may be due to using `IEnumerable` which is non-generic and assumes `object` for the type.   Does using `IEnumerable<dynamic>` as the cast work?

Reply:
It does not ... it fails with a RuntimeBinderException whereas just IEnumerable fails with an InvalidCastException.

Reply:
<img src="https://media1.giphy.com/media/3o6wrebnKWmvx4ZBio/giphy.gif"/>